functions/index.js

```diff
@@ -255,91 +308,73 @@ const updateUser = (req, res) => {
+        canUnlike: req.body.canUnlike ||"",
     };
-    if (Post.post == undefined) {
+    if (Postuser.post == undefined) {
         return res.status(400).json({ error: "Missing require parameter" });
     }
+    console.log("Starting Upload...", Postuser.username);
 
-    console.log("Starting Upload...", newUser.username);
-
-    db.collections("posts")
-        .add(Post)
+    db.doc(`/posts/${Postuser.post}`)
+        .create(Postuser)
         .then((doc) => {
             return res.status(201).json({ message: "Post created successfully" });
+
         })
         .catch((error) => {
             res.status(500).json({ error: error.code });
         });
 };
 
 // Update User Post
-
+// funciona
 const updatedPost = (req, res) => {
-    const post = {
-        post: req.body.post,
-        tags: req.body.tags,
-        postId: req.body.postId,
+    const editpost = {
+        post: req.body.post ||"",
+        tags: req.body.tags ||"",
+        title: req.body.title ||"",
 
     };
-    if (post.post == undefined) {
+    const username = req.body.username;
+    if (editpost.post == undefined) {
         return res.status(400).json({ error: "Missing requirement" });
     }
 
-    console.log("Post edited", user.username);
-    db.doc(`/posts/${post.postId}`)
-        .get()
-        .then((Post) => {
-            if (!Post.exists) {
-                throw {
-                    code: "post/non-existent",
-                    error: new Error(),
-                };
-            } else {
-                const existingPost = doc.data();
-                const updatedPost = {
-                    post: post.post || existingPost.post,
-                    tags: post.tags || existingPost.tags,
-                };
-                return db.doc(`/posts/${post.postId}`).update(updatedPost);
-            }
-        })
-        .then(() => {
-            return res.satus(200).json({ message: "Post updated successfully" });
+    console.log("Post edited", username);
+    db.doc(`/posts/${editpost.post}`)
+        .update(editpost)
+        .then((doc) => {
+            return res.status(200).json({ message: "Post updated successfully" });
         })
         .catch((error) => {
-            console.error(error);
-            if (error.code === "post/non-existent") {
-                return res
-                    .status(400)
-                    .json({ username: "Post doesn't exist in the system" });
-            }
             res.status(500).json({ error: error.code });
-        });
-
+            });
 };
 
 //Delete a Post
-
+//funciona
 const deletePost = (req, res) => {
     const delet = {
-        postId: req.body.postId,
+        username: req.body.username,
+        title: req.body.title,
+        post: req.body.post ||"",
 
     };
-
-    db.doc(`/posts/${post.postId}`)
-        .delete()
-        .then(() => {
-            return res.satus(200).json({ message: "Post updated successfully" });
+    db.doc(`/posts/${delet.post}`)
```

Comment:
similar as when we update, receive a postId as parameter then use the ID to delete and not the post text